snippets/xsd-snippets.json

```diff
@@ -0,0 +1,16 @@
+{
+  "el": {
+    "description": "[el]ement",
+    "prefix": "el",
+    "body": [
+      "<xs:element name=\"${1:string}\" type=\"${2|xs:string,xs:decimal,xs:integer,xs:boolean,xs:date,xs:time|}\" default=\"${1:string}\"/>"
```

Comment:
XSD completion which is based on XSD xml.xsd provides already element and attribute for completion, what is the benefit of those snippets?

Reply:
Does completion provides all attributes mentioned in snipped as so quickly as snippet does?

Reply:
> Does completion provides all attributes mentioned in snipped as so quickly as snippet does?

It provides only required attributes. Having too many completion items is not helpfull. So we should implement a java filter to show xs:element only if it is allowed. 

> BTW, why there is no XSD choice when changing language in VS Code?

It is a good sample why it is not interesting to support snippets with vscode, xsd:choice is only available of xs:complexType, so it should appear only when completion is triggered inside xs:complexType. As XSD completion is based onXSD schema, it show xsd:choice only when it is allowed.

Reply:
> I wanna check how completion works.

Create a new file with `test.xsd`. XSD uses XML as language (and not XSD).

Reply:
> It provides only required attributes. Having too many completion items is not helpfull. So we should implement a java filter to show xs:element only if it is allowed.

Then I guess my request is invalid.

Reply:
Once you will creat ethis file,you can opencompletion and you will see snippets:

![image](https://user-images.githubusercontent.com/1932211/211261980-c1889832-fe5f-42f8-900a-93addc298bd6.png)

You will notice that there are few snippets and not the all snippets of lemminx. We filter the snippet available only when a file is empty and have XSD extension.

Reply:
Sounds great. :)

Reply:
If you open completion, you will see that xsd:choice is not avialable inside xsd:element:

![image](https://user-images.githubusercontent.com/1932211/211262337-b978c8d3-cc4e-48d0-ace8-847f97cb8b70.png) but it is available inside complexType:

![image](https://user-images.githubusercontent.com/1932211/211262440-cf66907e-ca7b-4202-9971-5ed80b1f3d62.png)